dspace-api/src/main/java/org/dspace/health/Core.java

```diff
@@ -0,0 +1,185 @@
+        long total_size = 0;
+
+        Collections.sort(col_bitSizes, new Comparator<Object[]>() {
+            @Override
+            public int compare(Object[] o1, Object[] o2) {
+                try {
+                    return CollectionDropDown.collectionPath((Collection) o1[0]).compareTo(
+                        CollectionDropDown.collectionPath((Collection) o2[0])
+                    );
+                } catch (Exception e) {
+                    ret.append(e.getMessage());
+                }
+                return 0;
+            }
+        });
+        for (Object[] row : col_bitSizes) {
+            Long size = (Long) row[1];
+            total_size += size;
+            Collection col = (Collection) row[0];
+            ret.append(String.format(
+                    "\t%s:  %s\n", CollectionDropDown.collectionPath(col), FileUtils.byteCountToDisplaySize((long) size)));
+        }
+        ret.append(String.format(
+                "Total size:              %s\n", FileUtils.byteCountToDisplaySize(total_size)));
+
+        ret.append(String.format(
+                "Resource without policy: %d\n", getBitstreamsWithoutPolicyCount()));
+
+        ret.append(String.format(
+                "Deleted bitstreams:      %d\n", getBitstreamsDeletedCount()));
+
+        String list_str = "";
+        List<UUID> bitstreamOrphans = getBitstreamOrphansRows();
+        for (UUID id : bitstreamOrphans) {
+            list_str += String.format("%d, ", id);
+        }
+        ret.append(String.format(
+                "Orphan bitstreams:       %d [%s]\n", bitstreamOrphans.size(), list_str));
+
+        return ret.toString();
+    }
+
+    public  String getObjectSizesInfo() throws SQLException {
+        String ret = "";
+        for (String tb : new String[] { "Bitstream", "Bundle", "Collection",
+            "Community", "MetadataValue", "EPerson", "Item", "Handle",
+            "Group", "BasicWorkflowItem", "WorkspaceItem", }) {
+            int count = count("SELECT COUNT(*) from " + tb);
+            ret += String.format("Count %-14s: %s\n", tb,
+                String.valueOf(count));
+        }
+        return ret;
+    }
+
+
+    // get objects
+    //
+
+    public  List<Object[]> getWorkspaceItemsRows() throws SQLException {
+        return query("SELECT wi.stageReached, count(*) as cnt from WorkspaceItem wi group by wi.stageReached order by wi.stageReached");
+    }
+
+    public  List<UUID> getBitstreamOrphansRows() throws SQLException {
+        return query("select bit.id from Bitstream bit where bit.deleted != true" +
+                " and bit.id not in (select bit2.id from Bundle bun join bun.bitstreams bit2)" +
+                " and bit.id not in (select com.logo.id from Community com)" +
+                " and bit.id not in (select col.logo.id from Collection col)" +
+                " and bit.id not in (select bun.primaryBitstream.id from Bundle bun)");
+    }
```

Comment:
It seems wasteful to fetch the rows if all you want is to count them.  The DBMS can do that much more efficiently.  I would add new methods as required.  The point of a DAO is to get what you need from the DBMS and give it to you in the form you require without you having to know anything about how one gets it.  So DAOs tend over time to sprout many, many similar methods which correspond to specific queries.

I would think that a grouped result should be returned as a collection (or its iterator).  Again, I would let the DBMS do as much of the work as is practical, because it knows how the attribute values are stored and can compute efficient methods for summarizing them, and it is closer to the data.